File: Week 02/id_343/LeetCode_94.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func inorderTraversal(root *TreeNode) []int {
    if root == nil {
        return nil
    }
    l := inorderTraversal(root.Left)
    r := inorderTraversal(root.Right)
    return append(append(l, root.Val), r...)
}

File: Week 02/id_113/LeetCode94.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        traverse_path = []
        if root is None:
            return []

        traverse_path += self.inorderTraversal(root.left)
        traverse_path.append(root.val)
        traverse_path += self.inorderTraversal(root.right)

        return traverse_path